proto/server.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package proto;
+
+// NotaryServer defines the TUF operations that the server supports
+service Notary {
+	rpc Auth(Token) returns (AuthResponse) {}
+	rpc Update(Updates) returns (UpdateResponse) {}
+	rpc Get(RepoRole) returns (Repository) {}
+	rpc GetKey(RepoRole) returns (KeyResponse) {}
+}
+
+// Token holds a JWT that the Notary server will check for permissions
+message Token {
+	bytes jwt = 1;
+}
+
+message Updates {
+	bytes jwt = 1;
+	string gun = 2;
+	Repository repo = 3;
+}
+
+message RepoRole {
+	bytes jwt = 1;
```

Comment:
I am not sure if this can be changed, but it might be better to use a header/trailer to transmit the token, rather than including it on very request object. This way, we can use a credentials implementation to actually check it on each request. I am not yet sure how this should be implemented.